cpp/locality/NeighborQuery.h

```diff
@@ -51,6 +51,20 @@ struct QueryArgs
 {
     QueryArgs() = default;
 
+   //! Constructor
+    QueryArgs(QueryType mode, unsigned int num_neighbors, float r_max, float r_min, float r_guess,
+              float scale, bool exclude_ii)
+        : mode(mode), num_neighbors(num_neighbors), r_max(r_max), r_min(r_min), r_guess(r_guess),
+          scale(scale), exclude_ii(exclude_ii)
+    {}
+
+    //! Ball query factory
+    static QueryArgs make_ball(float r_max, float r_min = 0, bool exclude_ii = DEFAULT_EXCLUDE_II)
+    {
+        return QueryArgs(QueryType::ball, DEFAULT_NUM_NEIGHBORS, r_max, r_min, DEFAULT_R_GUESS, DEFAULT_SCALE,
+                         exclude_ii);
+    }
+
```

Comment:
I think these changes can be reverted because they're no longer being used. @DomFijan You refactored the calculation to include all point-point distances, including those greater than the largest `r_max = L_min/2` permitted by the box. Can you comment on how you justified that change? Theory? Literature? Matching to other expected results? We should add a comment explaining that in the C++ code and/or Python docstrings. I'll revert the changes in this file because they're no longer needed.